packages/client/src/utils/index.ts

```diff
@@ -1,8 +1,38 @@
+import { matchPath } from "react-router";
+
 import { topics } from "../config";
+import topicsJson from "../topics.json";
+
+// Get slug from url
+export const getSlugFromPathname = (pathname: string) => {
+  const match = matchPath(pathname, {
+    path: "/:slug",
+  }) as any;
+
+  return match?.params?.slug;
+};
 
 // Find a topic by the slug
-export const findTopicBySlug = (slug: string) =>
-  topics.find((t) => t.slug === slug);
+export const findTopicBySlug = (slug: string) => {
+  const topic = topics.find((t) => t.slug === slug);
+  if (topic) return topic;
+
+  const topicConfig = topicsJson.flat().find((topic) => topic.slug === slug);
+  if (!topicConfig) {
+    return null;
+  }
+
+  const name = topicConfig.name || topicConfig.slug;
+
+  return {
+    slug,
+    name,
+    hasIMTR: true,
```

Comment:
This is probably needed for the 'other checkers' coming from the Builder?

Reply:
correct, the 'dynamic' ones.